integration-tests/test/node_client.py

```diff
@@ -0,0 +1,196 @@
+        current_container_id = get_current_container_id()
+        current_container = docker_client.containers.get(current_container_id)
+        network = docker_client.networks.get(network_name)
+        network.connect(current_container)
+        network.reload()
+        container_network_config =network.attrs['Containers'][current_container.id]
+        ip, _ = container_network_config['IPv4Address'].split('/')
+        yield ip
+
+        network.disconnect(current_container)
+    else:
+        network_attr = docker_client.networks.get(network_name).attrs
+        ipam_attr = network_attr.get("IPAM")
+        assert ipam_attr is not None
+        ip = ipam_attr['Config'][0]['Gateway']
+        yield ip
+
+
+def get_current_container_id() -> str:
+    hostname = subprocess.run(['hostname'], stdout=subprocess.PIPE)
+    return hostname.stdout.decode('utf8').strip("\n")
+
+
+class TransportServer(TransportLayerServicer):
+    def __init__(self, node: Node, networkId: str, return_queue: Queue):
+        self.node = node
+        self.header = Header(sender=self.node, networkId=networkId)
+        self.return_queue = return_queue
+        super(TransportServer, self).__init__()
+
+    def Send(self, request: TLRequest, context: grpc.ServicerContext) -> TLResponse:
+        return TLResponse(noResponse=NoResponse(header=self.header))
+
+    def Stream(self, request_iterator: Iterator[Chunk], context: grpc.ServicerContext) -> ChunkResponse:
+        message_cls = None
+        data = b''
+        for chunk in request_iterator:
+            content_type = chunk.WhichOneof('content')
+            if content_type == 'header':
+                typeId = chunk.header.typeId
+                message_cls = getattr(CasperMessage_pb2, typeId)
+            elif content_type == 'data':
+                data = chunk.data.contentData
+            else:
+                raise NotImplementedError()
+        assert message_cls is not None
+        stream_message = message_cls()
+        stream_message.ParseFromString(data)
+        self.return_queue.put(stream_message)
+        return ChunkResponse()
+
+
+class NodeClient:
+    def __init__(self, node_pem_cert: bytes, node_pem_key: bytes, host: str, network_name: str, network_id: str = DEFAULT_NETWORK_ID):
+        self.node_pem_cert = node_pem_cert
+        self.node_pem_key = node_pem_key
+        self.ec_key = load_pem_private_key(self.node_pem_key, password=None, backend=default_backend())
+        self.network_id = network_id
+
+        self.host = host
+        self.tcp_port = 0
+        self.udp_port = get_free_tcp_port()
+        self.network_name = network_name  # docker network name
+
+        self.return_queue = Queue() # type: ignore
+
+        self.server = self._start_transport_server()
+
+    @property
+    def node_pb(self) -> Node:
+        return Node(id=get_node_id_raw(self.ec_key), host=self.host.encode('utf8'), tcp_port=self.tcp_port, udp_port=self.udp_port)
```

Comment:
Can `id=get_node_id_raw(self.ec_key)` be taken out into a separate line please?  It's gonna be easier to step over in the debugger.